lib/scope.js

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+function Scope(arr) {
+  if (!(this instanceof Scope)) {
+    return new Scope(arr);
+  }
+  if (!arr) {
+    arr = [];
+  } else if (arr instanceof Scope) {
+    return arr;
+  } else if (typeof arr === 'string') {
+    arr = arr.split(' ');
```

Comment:
I'm saddened that `o[''] = 5` actually works, and is included in `Object.keys(o)`